internal/app/cloudwatcher/node_stats.go

```diff
@@ -0,0 +1,458 @@
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitBytes,
+			},
+			Selector: func(ns *NodeStats) *float64 {
+				f := float64(ns.JVMHeapStats.MaxBytes)
+				return &f
+			},
+		},
+
+		&StatsData{
+			Template: cloudwatch.MetricDatum{
+				MetricName:        aws.String("JVMUsedBytes"),
+				Timestamp:         aws.Time(now),
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitBytes,
+			},
+			Selector: func(ns *NodeStats) *float64 {
+				f := float64(ns.JVMHeapStats.UsedBytes)
+				return &f
+			},
+		},
+
+		&UtilizationData{
+			Template: cloudwatch.MetricDatum{
+				MetricName:        aws.String("JVMUtilization"),
+				Timestamp:         aws.Time(now),
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitPercent,
+			},
+			Numerator: func(ns *NodeStats) *float64 {
+				f := float64(ns.JVMHeapStats.UsedBytes)
+				return &f
+			},
+			Denominator: func(ns *NodeStats) *float64 {
+				f := float64(ns.JVMHeapStats.MaxBytes)
+				return &f
+			},
+		},
+
+		&StatsData{
+			Template: cloudwatch.MetricDatum{
+				MetricName:        aws.String("FSTotalBytes"),
+				Timestamp:         aws.Time(now),
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitBytes,
+			},
+			Selector: func(ns *NodeStats) *float64 {
+				if !ns.HasRole("data") {
+					return nil
+				}
+				f := float64(ns.FilesystemTotalBytes)
+				return &f
+			},
+		},
+
+		&StatsData{
+			Template: cloudwatch.MetricDatum{
+				MetricName:        aws.String("FSAvailableBytes"),
+				Timestamp:         aws.Time(now),
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitBytes,
+			},
+			Selector: func(ns *NodeStats) *float64 {
+				if !ns.HasRole("data") {
+					return nil
+				}
+				f := float64(ns.FilesystemAvailableBytes)
+				return &f
+			},
+		},
+
+		&UtilizationData{
+			Template: cloudwatch.MetricDatum{
+				MetricName:        aws.String("FSUtilization"),
+				Timestamp:         aws.Time(now),
+				Dimensions:        dimensions,
+				StorageResolution: aws.Int64(1),
+				Unit:              cloudwatch.StandardUnitPercent,
+			},
+			Numerator: func(ns *NodeStats) *float64 {
+				if !ns.HasRole("data") {
+					return nil
+				}
+				f := float64(ns.FilesystemTotalBytes) - float64(ns.FilesystemAvailableBytes)
+				return &f
+			},
+			Denominator: func(ns *NodeStats) *float64 {
+				if !ns.HasRole("data") {
+					return nil
+				}
+				f := float64(ns.FilesystemTotalBytes)
+				return &f
+			},
+		},
+	}
+
+	pools := make(map[string]struct{}, 3)
+	for _, ns := range s {
+		for pool := range ns.JVMHeapPools {
+			if _, ok := pools[pool]; !ok {
+				func(pool string) { // Make sure refernce to pool isn't shadowed.
```

Comment:
What is a "shadowed" reference and why would it happen here?

Reply:
I think I used the wrong term here. [This](https://github.com/golang/go/wiki/CommonMistakes#using-goroutines-on-loop-iterator-variables) is the problem I meant to avoid.

If anything, I want the variable to be [shadowed](https://yourbasic.org/golang/gotcha-shadowing-variables/).

- [ ] Fix shadowed variable comment.